app/src/main/java/org/mozilla/focus/SettingsFragment.kt

```diff
@@ -94,9 +95,11 @@ class SettingsFragment : Fragment() {
         // cleared from this setting and nothing is selected. This is fine: the user can press
         // left-right to focus something else and it's an edge case that I don't think it is worth
         // adding code to fix.
-        val shouldFocus = !context.isVoiceViewEnabled()
-        telemetryView.isFocusable = shouldFocus
-        if (!shouldFocus) { telemetryView.clearFocus() }
+        context?.let {
```

Comment:
I don't love `?.let`: it's very easy to `.let` and break your program.

In this case, I think I'd prefer:
```kotlin
val context = context ?: return
...
```

Or something like that.

Reply:
I do not really understand how `.let` will break the program. In this case the compiler would again complain that you are dealing with a nullable inside the block?

But I agree that your approach is nicer to read in this case. I'll update!

Reply:
But why would I use `let` without using the object inside the block?

Reply:
Two cases I'd been concerned about are the code being refactored or a developer just accidentally starts using the optional type without thinking about it, e.g.:
```kotlin
maybeBrowserFragment.let {
    val webView = it?.webView
    return webView?.canGoBack() ?: false
}
```

I think this isn't that extreme of an example but I have hesitations it can get more confusing.

I think I also dislike let because its intentions are unclear. It stands in for multiple features that already exist:
- let without ? is reassignment, i.e. `val whatever = <let value>`
- let with ? is casting to non-optional types

One advantage it does have is that it groups together related concepts (with a scope; I'm assuming it actually returns a value though), e.g.:
```kotlin
val firstActiveToolbarButton = browserFragment?.let {
    val toolbar = it?.toolbar
    toolbar.children.firstOrNull { it.isActive() }
}
// Unrelated code.
```

But I think putting newlines between related concepts in code or pushing this functionality to helper functions does this just as well.

So overall, it feels like let is lacking intention and is redundant. I'm also concerned with how refactoring `let`s in and out of the code destroys blame (a problem I'm also having with `apply` and `also`) even if I do actually really like their conceptual groupings, assuming others are using them like I do.

I could be wrong about this, let me know what you think! :)

@colintheshots I'm wondering if you have an opinion on `let` too! :)